### PR TITLE
Load X509 cert on client creation

### PR DESCRIPTION
**Pull request by tokenclt**
This PR proposes a fix for issue #111. Changes are:
* Execute `LoadX509Cert` when creating `Client`, instead of making requests.
* Prefer to use the provided `Transport` (from `httpClient` argument  of `NewClient`), only create a new `Transport` when user sets `Cert`, `Key` and `Insecure`.
* Fix some go vet suggestions.

BTW, I've submitted the Apache CLA and I'm waiting for the confirmation email.

### CODE REVIEW

**Comment by csantanapr** [comment]
Thank you @AprliRainkun 👍 

@dubee or @mdeuser could you review as your earliest convenience 🙏

File: whisk/client.go [hunk 1]
```diff
@@ -176,17 +182,28 @@ func NewClient(httpClient *http.Client, config_input *Config) (*Client, error) {
...
 	return c, nil
 }
 
 func (c *Client) LoadX509KeyPair() error {
+	var tlsModified bool
+
 	tlsConfig := &tls.Config{
 		InsecureSkipVerify: c.Config.Insecure,
 	}
+	// reasoned from: tlsModified = (c.config.Insecure != boolean default)
```
Comment (rabbah) [inline whisk/client.go]: i think it's ok to remove these 2 comment lines

File: whisk/client.go [hunk 2]
```diff
@@ -214,19 +231,26 @@ func (c *Client) LoadX509KeyPair() error {
...
-			KeepAlive: 30 * time.Second,
-			DualStack: true,
-		}).DialContext,
-		MaxIdleConns:          100,
-		IdleConnTimeout:       90 * time.Second,
-		TLSHandshakeTimeout:   10 * time.Second,
-		ExpectContinueTimeout: 1 * time.Second,
-		TLSClientConfig:       tlsConfig,
+	// only set new transport when the TLS config is nondefault
+	if tlsModified {
```
Comment (mdeuser) [inline whisk/client.go]: maybe no need to introduce a new var (tlsModified) that's set in two places.. instead just
```
// Only replace the existing transport when a custom TLS configuration is needed
if c.Config.Insecure || tlsConfig.Certificates != nil {
```

Comment (mdeuser) [reply whisk/client.go]: even better!

**Pull request merged**